kernel/sql-federation/core/src/main/java/org/apache/shardingsphere/sqlfederation/engine/SQLFederationEngine.java

```diff
@@ -113,26 +113,14 @@ public SQLFederationEngine(final String databaseName, final String schemaName, f
      * @param globalRuleMetaData global rule meta data
      * @return use SQL federation or not
      */
-    @SuppressWarnings({"unchecked", "rawtypes"})
     public boolean decide(final SQLStatementContext sqlStatementContext, final List<Object> parameters,
                           final ShardingSphereDatabase database, final RuleMetaData globalRuleMetaData) {
         // TODO BEGIN: move this logic to SQLFederationDecider implement class when we remove sql federation type
         if (isQuerySystemSchema(sqlStatementContext, database)) {
             return true;
         }
         // TODO END
-        boolean sqlFederationEnabled = sqlFederationRule.getConfiguration().isSqlFederationEnabled();
```

Comment:
Why change this logic?

Reply:
Hi mentor, while ensuring the correct execution of the decide method, this is a small change I have attempted.
The code below is the decision logic for making query statements. Place this part of the logic in a method named selectDecide and change the judgment logic of code `!sqlFederationEnabled || !(sqlStatementContext instanceof SelectStatementContext)` to code `isEnableSqlFederation() && selectDecide(sqlStatementContext, parameters, database, globalRuleMetaData).`
I think it will be easier to understand the decision logic of the decide method.